src/components/bottom-widget.js

```diff
@@ -38,23 +39,105 @@ const propTypes = {
 
 const maxWidth = 1080;
 
-BottomWidgetFactory.deps = [TimeWidgetFactory, AnimationControlFactory];
-
 const BottomWidgetContainer = styled.div`
   position: absolute;
   display: flex;
   flex-direction: column;
-  padding-top: ${props => props.theme.bottomWidgetPaddingTop}px;
-  padding-right: ${props => props.theme.bottomWidgetPaddingRight}px;
-  padding-bottom: ${props => props.theme.bottomWidgetPaddingBottom}px;
-  padding-left: ${props => props.theme.bottomWidgetPaddingLeft}px;
+  padding-top: ${props => (props.hasPadding ? props.theme.bottomWidgetPaddingTop : 0)}px;
+  padding-right: ${props => (props.hasPadding ? props.theme.bottomWidgetPaddingRight : 0)}px;
+  padding-bottom: ${props => (props.hasPadding ? props.theme.bottomWidgetPaddingBottom : 0)}px;
+  padding-left: ${props => (props.hasPadding ? props.theme.bottomWidgetPaddingLeft : 0)}px;
   width: ${props => props.width}px;
   bottom: 0;
   right: 0;
   z-index: 1;
 `;
 
-export default function BottomWidgetFactory(TimeWidget, AnimationControl) {
+FilterAnimationControllerFactory.deps = [AnimationControllerFactory];
+export function FilterAnimationControllerFactory(AnimationController) {
+  const FilterAnimationController = ({
+    filter = {},
+    filterIdx,
+    setFilterAnimationTime,
+    children
+  }) => {
+    const bins = useMemo(() => filter && filter.bins, [filter]);
+    const interval = useMemo(() => filter.plotType && filter.plotType.interval, [filter]);
+    const intervalBins = useMemo(
+      () => bins && Object.keys(bins).length && Object.values(bins)[0][interval],
+      [bins, interval]
+    );
```

Comment:
since you are using bins and interval only to compute intervalBins you can do the following:
```
const intervalBins = useMemo(() => {
    const {bins, plotType} = filter;
    const interval = plotType && plotType.interval);
    const intervalBins = bins && Object.keys(bins).length && Object.values(bins)[0][interval];
},[filter])
```